src/FSharp.CodeFormat/CodeFormatAgent.fs

```diff
@@ -181,87 +196,88 @@ type CodeFormatAgent() =
-                    |> Option.getOrElse (leftPoint+1, rightPoint, TokenKind.String))
+                    |> Array.tryFind (fun (range,category) -> range.StartColumn = leftPoint) 
+                    |> Option.bind (fun (range,category)->
+                         categoryToTokenKind category
+                         |> Option.map (fun kind -> range.StartColumn, range.EndColumn, kind))
+                    |> Option.defaultValue (leftPoint+1, rightPoint, TokenKind.String))
               
               for leftPoint, rightPoint, kind in data do
-                yield Token (kind, lineStr.[leftPoint..rightPoint-1], None)
+                yield FSharp.CodeFormat.Token (kind, lineStr.[leftPoint..rightPoint-1], None)
           // Process the rest of the line
           yield! loop island rest stringRange }
 
     // Process the current line & return info about it
     Line (loop [] (List.ofSeq lineTokens) None |> List.ofSeq)
 
   /// Process snippet
-  let processSnippet checkResults categorizedSpans lines (snippet: Snippet) =
+  let processSnippet checkResults categorizedRanges lines (snippet: Snippet) =
     snippet 
     |> List.map (fun snippetLine ->
         processSnippetLine 
             checkResults 
-            (categorizedSpans 
+            (categorizedRanges 
              |> Map.tryFind ((fst snippetLine).StartLine + 1) 
-             |> function None -> [] | Some spans -> List.ofSeq spans) 
+             |> function None -> [||] | Some spans -> Array.ofSeq spans) 
             lines 
             ((fst snippetLine).StartLine, snd snippetLine))
 
 // --------------------------------------------------------------------------------------
 
   // Create an instance of an InteractiveChecker (which does background analysis
   // in a typical IntelliSense editor integration for F#)
-  let languageService = LanguageService()
-  do languageService.SetCriticalErrorHandler(fun exn str1 str2 opts ->
-      FSharp.Formatting.Common.Log.errorf "Language Service Error (%s, %s, %A): %O" str1 str2 opts exn)
+  let fsChecker = FSharpChecker.Create()
+  //do fsChecker.SetCriticalErrorHandler(fun exn str1 str2 opts ->
+  //    FSharp.Formatting.Common.Log.errorf "Language Service Error (%s, %s, %A): %O" str1 str2 opts exn)
 
   /// Type-checking takes some time and doesn't return information on the
   /// first call, so this function creates workflow that tries repeatedly
-  let getTypeCheckInfo(file, source, opts) = async {
-      let! checkResults = languageService.ParseAndCheckFileInProject(opts, file, source, AllowStaleResults.No)
-      let! symbolUses = languageService.GetAllUsesOfAllSymbolsInFile (opts, file, source, AllowStaleResults.No, false, new Profiler())
-      return checkResults, symbolUses
-  }
+  //let getTypeCheckInfo(filePath, source, opts) = asyncMaybe {
+      
+
+  //    let! symbolUses = checkResults.GetAllUsesOfAllSymbolsInFile () |> liftAsync
+  //    return checkResults, symbolUses
+  //}
```

Comment:
i'm still getting all of the build configuration kinks worked out, i'll come back around and polish it up all spiffy once it actually works again 😉